src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class FruitTransaction {
+    private final Operation operation;
+    private final String fruitName;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruitName, int quantity) {
+        this.operation = operation;
+        this.fruitName = fruitName;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruitName() {
+        return fruitName;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String operation;
+
+        Operation(String operation) {
+            this.operation = operation;
+        }
+
+        public String getOperation() {
+            return operation;
+        }
+
+        public static FruitTransaction.Operation getOperationByValue(String operation) {
+            return Arrays.stream(Operation.values())
+                    .filter(o -> o.getOperation().equals(o.operation))
```

Comment:
There is a logical error in the lambda expression. It should compare the passed 'operation' parameter with the 'operation' field of the enum, not the 'operation' field with itself. Change `o.getOperation().equals(o.operation)` to `o.getOperation().equals(operation)`.